app/src/main/java/com/duckduckgo/app/browser/omnibar/Omnibar.kt

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright (c) 2024 DuckDuckGo
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.duckduckgo.app.browser.omnibar
+
+import android.annotation.SuppressLint
+import android.content.res.TypedArray
+import android.view.View
+import androidx.coordinatorlayout.widget.CoordinatorLayout
+import androidx.core.view.updateLayoutParams
+import com.duckduckgo.app.browser.databinding.FragmentBrowserTabBinding
+import com.duckduckgo.app.browser.omnibar.model.OmnibarPosition
+import com.duckduckgo.mobile.android.R as CommonR
+
+@SuppressLint("ClickableViewAccessibility")
+class Omnibar(
+    val omnibarPosition: OmnibarPosition,
+    private val binding: FragmentBrowserTabBinding,
+    private val omnibarPositionFeature: ChangeOmnibarPositionFeature,
+) {
+    private val actionBarSize: Int by lazy {
+        val array: TypedArray = binding.rootView.context.theme.obtainStyledAttributes(intArrayOf(android.R.attr.actionBarSize))
+        val actionBarSize = array.getDimensionPixelSize(0, -1)
+        array.recycle()
+        actionBarSize
+    }
+
+    val appBarLayout: LegacyOmnibarView by lazy {
+        if (omnibarPosition == OmnibarPosition.TOP || !omnibarPositionFeature.self().isEnabled()) {
+            binding.rootView.removeView(binding.legacyOmnibarBottom)
```

Comment:
why remove the view instead of making it gone?

Reply:
This is a remnant of the previous implementation when we used included layouts. Both top and bottom would use the same view IDs so one had to be removed. I’ll make the change to hide it.

Reply:
Actually, after some testing it turns out this messes up the top omnibar. I think there are still ID conflicts, because it works fine when the views are removed. Reverting.

https://github.com/user-attachments/assets/06f91d82-fae4-4fc0-a67e-b86423cab981